### PR TITLE
fix: nonce increment/decrement functionality using arrow buttons

### LINKED ISSUE #23535: [Bug]: Customizing nonce with increment buttons adjusts nonce value to 0 or 1.
### Describe the bug

When customizing nonce using lower increment button nonce value sets to 0, using higher increment button results in nonce = 1.

### Expected behavior

Increment buttons should set suggested nonce to +1 or -1.

### Screenshots/Recordings

https://github.com/MetaMask/metamask-extension/assets/104780023/a58db280-ef2f-450f-b630-1e34336832c9

### Steps to reproduce

1. Go to 'Settings' -> 'Advanced'
2. Turn 'Customize transaction nonce' toggle ON
3. Start a 'Send' transaction
4. Try to use increment buttons to customize transaction nonce on confirmation page

### Error messages or log output

_No response_

### Version

11.10.1, 11.13.0

### Build type

None

### Browser

Chrome, Firefox

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**

**Context**
When customizing the nonce using a lower increment button nonce value is set to 0, using higher increment button results in nonce = 1.

This PR aims to fix the support of the arrow buttons to increase/decrease the number taking the suggested next nonce as the base. 

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26569?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/23535

## **Manual testing steps**

1. Enable custom nonce
2. Send Eth
3. Customize nonce using the arrow buttons

### **After**

[Screencast from 21-08-2024 10:42:17.webm](https://github.com/user-attachments/assets/d4fc505f-e2f0-4636-8462-5f0c1a2e050b)

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by sleepytanya** [comment]
Android build  https://app.bitrise.io/build/a43b7409-29ed-4de1-88cc-02d3eafb5b46?tab=artifacts
iOS build  https://app.bitrise.io/build/f44cdebe-1cd2-4096-9f76-5cf0dc099286

Nonce values can be adjusted with an increment / decrement buttons:


iOS

https://github.com/user-attachments/assets/400600ab-e8af-4b5c-a35c-a9d305560111

Android



https://github.com/user-attachments/assets/8421f896-fbcb-4dca-9502-45d1a53c7747

File: ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js [hunk 1]
```diff
@@ -431,6 +430,19 @@ export default class ConfirmTransactionBase extends Component {
...
 
+    const handleNonceChange = ({ target: { value } }) => {
+      const inputValue = Number(value);
+
+      if (inputValue < 0 || isNaN(inputValue)) {
+        updateCustomNonce('');
+        return;
+      }
+
+      updateCustomNonce(String(inputValue));
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js:441]: Assuming the `inputValue` is a decimal and not hexadecimal, why are we storing it as a string?

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

File: ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js (hunk 1 above)
Comment (vinistevam) [reply ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js:441]: Good point, no need to store it as a string, removed.

File: ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js [hunk 2]
```diff
@@ -431,6 +430,19 @@ export default class ConfirmTransactionBase extends Component {
       );
     };
 
+    const handleNonceChange = ({ target: { value } }) => {
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js:433]: 100% we need to use the `usaCallback`, but in this case, it is an old class component.

File: test/e2e/tests/tokens/increase-token-allowance.spec.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ describe('Increase Token Allowance', function () {
         fixtures: new FixtureBuilder()
           .withPermissionControllerConnectedToTestDapp()
           .build(),
-        ganacheOptions: defaultGanacheOptions,
+        ganacheOptions: multipleGanacheOptions,
```
Comment (vinistevam) [inline test/e2e/tests/tokens/increase-token-allowance.spec.js:28]: The pipeline was failing with no funds in `ACCOUNT_2`

**Comment by sleepytanya** [comment]
Works great!
 


https://github.com/user-attachments/assets/c43b897f-c3b6-472d-ad2b-03ea54eaa501

**Review by pedronfigueiredo — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**